server.js

```diff
@@ -1,49 +1,41 @@
-let categories = require('./my-app-react/src/data/categories.json')
-let products = require('./my-app-react/src/data/products.json')
+let dataService = require('./dataservice.js');
 
-const express = require('express')
-const app = express()
+const express = require('express');
+const app = express();
 
 app.get('/info', (req, res) => {
-    info = [
-        {"serverName": "localhost5000"},
+    let info = [
+        {"serverName": "localhost5001"},
         {"sampleName": "chinemerem-react"},
         {"serverVersion": "1.0.0"}
-    ]
+    ];
     res.json(info)
-})
+});
 
 app.get('/products/all', (req, res) => {
-    res.json(products)
-})
+    let productsMap = dataService.getCombinedProductMap();
+    res.send(productsMap);
+});
 
 app.get('/products/:id', (req, res) => {
-    let urlProductID = req.params.id
-    let productsLocal = products.products
-
-    for (productID in productsLocal) {
-        let product = productsLocal[productID];
-        if(urlProductID == product.id) {
-            res.json(product)
-        }
-    }
-
-})
+    let urlProductID = req.params.id;
+    let productsMap = dataService.getCombinedProductMap();
+    
+    res.send(productsMap.get(urlProductID));
+});
 
 app.get('/category/:id', (req, res) => {
-    let urlCategoryID = req.params.id
-    let productsLocal = products.products
-    let arrayOfProducts = []
-
-    for (productID in productsLocal) {
-        let product = productsLocal[productID];
-        if(urlCategoryID == product.categoryId) {
-            arrayOfProducts.push(product)
+    let urlCategoryID = req.params.id;
+    let combinedProductMap = dataService.getCombinedProductMap();
+    const productsBasedOnCategory = new Map();
+    combinedProductMap.forEach(((value, key, map) => {
+        if (value.categoryId === urlCategoryID) {
+            productsBasedOnCategory.set(key, value);
         }
-    }
+    }));
     
-    res.json(arrayOfProducts)
-})
+    res.send(productsBasedOnCategory);
```

Comment:
The output is in the wrong format. It should be

```json
{
  "P001": {
    "name": "...",
    "categoryId": "...",
    ...
  },
  "P002": {
    "name": "...",
    "categoryId": "...",
    ...
  }
}
```

Reply:
I told you it was rendered as `{ }` in my browser.